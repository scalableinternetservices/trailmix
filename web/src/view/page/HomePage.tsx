import { RouteComponentProps } from '@reach/router'
import * as React from 'react'
import { ColorName, Colors } from '../../../../common/src/colors'
import { H1, H2, H3 } from '../../style/header'
import { Spacer } from '../../style/spacer'
import { style } from '../../style/styled'
import { BodyText } from '../../style/text'
import { AppRouteParams } from '../nav/route'
import { Page } from './Page'

interface HomePageProps extends RouteComponentProps, AppRouteParams {}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export function HomePage(props: HomePageProps) {
  return (
    <Page>
      <Hero>
        <H1>Trailmix</H1>
        <H3>Hike and Have Fun!</H3>
      </Hero>
      <Content>
        <LContent>
          <Section>
            <H2>About Trailmix</H2>
            <Spacer $h4 />
            <BodyText>A hiking application to make finding hikes easier and fun!</BodyText>
            <Spacer $h4 />
            <BodyText>
              Instructions: Input zip code -- get hikes nearby -- go on hikes! -- leave comments with your feedback!
            </BodyText>
          </Section>
          <Section>
            <img
              className="mw-50"
              src="https://cdn.pixabay.com/photo/2015/03/26/09/52/hiking-690479_960_720.jpg"
              alt="logo"
            />
          </Section>
        </LContent>
        <RContent>
          <Section>
            <H2>Team Information</H2>
            <Spacer $h4 />
            <BodyText>
              <table>
                <tbody>
                  <tr>
                    <TD>👨‍💻</TD>
                    <TD>Jivan Gubbi</TD>
                    <TD>👩‍💻</TD>
                    <TD>Hayley Martinez</TD>
                  </tr>
                  <tr>
                    <TD>👩‍💻</TD>
                    <TD>Shikha Mody</TD>
                    <TD>👨‍💻</TD>
                    <TD>Armaan Singh</TD>
                  </tr>
                </tbody>
              </table>
            </BodyText>
          </Section>
          <Section>
            <BodyText>Scalable Internet Services</BodyText>
            <Spacer $h4 />
            <BodyText>UCLA, Fall 2020</BodyText>
          </Section>
        </RContent>
      </Content>
    </Page>
  )
}

const Hero = style('div', 'mb4 w-100 ba b--mid-gray br2 pa3 tc', {
  borderLeftColor: Colors.mint + '!important',
  borderRightColor: Colors.mint + '!important',
  borderLeftWidth: '4px',
  borderRightWidth: '4px',
})

const Content = style('div', 'flex-l')

const LContent = style('div', 'flex-grow-0 w-70-l mr4-l')

const RContent = style('div', 'flex-grow-0  w-30-l')

const Section = style('div', 'mb4 mid-gray ba b--mid-gray br2 pa3', (p: { $color?: ColorName }) => ({
  borderLeftColor: Colors[p.$color || 'mint'] + '!important',
  borderLeftWidth: '3px',
}))

const TD = style('td', 'pa1', p => ({
  color: p.$theme.textColor(),
}))
