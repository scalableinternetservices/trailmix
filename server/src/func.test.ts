import renderer from 'react-test-renderer'
import { TrailInfoCard } from '../../web/src/view/page/TrailInfo'
test('Math.random() returns value between 0 and 1', () => {
  expect(Math.random()).toBeGreaterThanOrEqual(0)
  expect(Math.random()).toBeLessThan(1)
})

const args = {
  title: 'Hollywood Sign Trail',
  description: 'Hike the Hollywood Sign!',
  onClick: () => console.log('CLICKED TRAIL'),
}
test('trail card renders correctly', () => {
  const tree = renderer.create(TrailInfoCard({ ...args })).toJSON()
  expect(tree).toMatchSnapshot()
})
