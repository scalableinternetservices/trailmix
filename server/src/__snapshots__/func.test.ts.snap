// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trail card renders correctly 1`] = `
<div
  className="flex items-center pa2 hover-bg-light-green bg-washed-green"
  id="trailInfo"
  onClick={[Function]}
  style={
    Object {
      "borderRadius": "25px",
      "opacity": 1,
      "outline": "none",
      "width": "70%",
    }
  }
>
  <img
    className="ph3"
  />
  <div
    className="flex flex-column"
  >
    <div>
      <b
        className="pv2 "
      >
        Hollywood Sign Trail
      </b>
      <b
        className="pv2 "
      >
        
      </b>
    </div>
    <b
      className="pv2 "
    >
      Trail Length: undefined miles Difficulty: undefined
    </b>
    <b
      className="pb2 "
    >
      Hike the Hollywood Sign!
       
    </b>
  </div>
</div>
`;
